aten/src/ATen/native/Loss.cpp

```diff
@@ -463,35 +455,10 @@ Tensor& mse_loss_backward_out(const Tensor& grad_output,
 }
 
 Tensor l1_loss(const Tensor& input, const Tensor& target, int64_t reduction) {
-  const auto float_type = c10::toValueType(input.scalar_type());
-  Tensor result = at::empty({0}, input.options().dtype(float_type));
-  return at::l1_loss_out(result, input, target, reduction);
-}
-
-Tensor& l1_loss_out(const Tensor& input, const Tensor& target, int64_t reduction, Tensor& result) {
-  if (reduction != Reduction::None) {
-    auto diff = at::sub(input, target);
-    auto loss = diff.is_complex() ? diff.abs() : diff.abs_();
-    if (reduction == Reduction::Mean) {
-      return at::mean_out(result, loss, IntArrayRef{});
-    } else {
-      return at::sum_out(result, loss, IntArrayRef{});
-    }
-  } else {
-    auto diff = input.is_complex() ? at::sub(input, target) : at::sub_out(result, input, target);
-    return at::abs_out(result, diff);
-  }
-}
-
-Tensor l1_loss_backward(const Tensor& grad_output, const Tensor& input, const Tensor& target, int64_t reduction) {
-  Tensor grad_input = at::zeros_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  return at::l1_loss_backward_out(grad_input, grad_output, input, target, reduction);
-}
-
-Tensor& l1_loss_backward_out(const Tensor& grad_output,
-    const Tensor& input, const Tensor& target, int64_t reduction, Tensor& grad_input) {
-  auto norm = reduction == Reduction::Mean ? grad_output / input.numel() : grad_output;
-  return at::sub_out(grad_input, input, target).sgn_().mul_(norm);
+  auto diff = input - target;
+  auto output = diff.is_complex() ? at::real(diff).abs() + at::imag(diff).abs()
+                                  : diff.abs();
```

Comment:
If `v` is a complex or a real vector, the L1 norm is then `||v||_1 = \sum_{i=1}^n |v_i|`, where `|v_i|` is the module of its i-th coordinate. In case of complex numbers it is `sqrt(real^2 + imag^2)`.